src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNormalizerTests.cs

```diff
@@ -5979,5 +5979,86 @@ public void TestNormalizeParseExpressionLiteralCharacter(string expression)
             var syntaxNode = SyntaxFactory.ParseExpression(expression).NormalizeWhitespace();
             Assert.Equal(expression, syntaxNode.ToFullString());
         }
+
+        [Fact]
+        public void TestNormalizeAllowsRefStructConstraint_01()
+        {
+            TestNormalizeDeclaration("""
+                class C1<T> where T:allows   ref   struct   ;
+                class C2<T, S> where T:allows   ref   struct,where S:struct     ;
+                class C3<T> where T:struct,allows   ref   struct            ;
+                class C4<T> where T:new(),allows   ref   struct          ;
+                class C5<T>
+                where
+                T
+                :
+                allows
+                ref
+                struct
+                ;
+                """, """
+                class C1<T>
+                  where T : allows ref struct;
+                class C2<T, S>
+                  where T : allows ref struct , where S : struct;
```

Comment:
The space before the comma looks unexpected. Shouldn't it be:
```suggestion
                  where T : allows ref struct, where S : struct;
```

Reply:
> The space before the comma looks unexpected. 

It is a syntax error, a comma cannot follow `allows ref struct` constraint